lib/puppet/provider/openssl.rb

```diff
@@ -0,0 +1,54 @@
+require 'etc'
+
+# class to use in openssl providers to handle file permission (mode, group and owner)
+class Puppet::Provider::Openssl < Puppet::Provider
+  include Puppet::Util::POSIX
+
+  def owner
+    if File.exist?(@resource[:path])
+      Etc.getpwuid(File.stat(@resource[:path]).uid).name
+    else
+      :absent
+    end
+  end
+
+  def owner=(should)
+    File.send(:chown, uid(should), nil, resource[:path])
```

Comment:
What's the benefit of `File.send(:chown)` over `File.chown` here?

Reply:
I do not know ! I'm not a ruby expert ;).

Reply:
Then I wonder how you got to this code. Normally I'd not expect to see `send` in code this way but rather:
```suggestion
    File.chown(uid(should), nil, resource[:path])
```

Reply:
ok. It's some time I wrote this (and sent to the former owner of this module...) so I cannot remember, where I found it !
I'll take your suggestion, and adapt at the other places...

Reply:
https://stackoverflow.com/questions/3337285/what-does-send-do-in-ruby has some explanation. Essentially it can be used for dynamic method calls (call a method with a variable name) or bypass visibility checks (call private methods). Both aren't relevant here which is why I didn't expect it here.